src/metabase/models/dashboard.clj

```diff
@@ -615,15 +615,28 @@
          :serdes/meta  [{:model "Dashboard"    :id (:entity_id dashboard)}
                         {:model "DashboardTab" :id (:entity_id tab)}]))
 
+
+(defn- drop-excessive-nested!
+  "Remove nested entities which are not present in incoming serialization load"
+  [entities model field value]
+  (let [known     (t2/select-fn-set :entity_id model field value)
+        to-remove (set/difference known (set (map :entity_id entities)))]
+    (when (seq to-remove)
+      (t2/delete! model :entity_id [:in to-remove]))))
```

Comment:
I wonder whether we can leverage hydration to do a lot of this for us. Since hydration is used to add the nested entities onto the extraction, we can assume that the nested entities are on the same hydration keys.

https://github.com/metabase/metabase/blob/abaf5e2d04f1b801e1722e07575ff42b6714a6f6/src/metabase/models/dashboard.clj#L580-L584
```clojure
(defn- drop-excessive-nested!
  "Delete nested local entities which are not present in incoming ingested entity."
  [ingested local hydration-key]
  (let [local-nested    (get (t2/hydrate local hydration-key) hydration-key)
        ingested-nested (get ingested hydration-key)
        to-remove       (set/difference (set (map :entity_id local))
                                        (set (map :entity_id ingested)))
        model           (t2/model (first local-nested))]
    (when (seq to-remove)
      (t2/delete! model :entity_id [:in to-remove]))))
```

Then we can call it in a much simpler way:
```
before
(drop-excessive-nested! (:tabs ingested) :model/DashboardTab :dashboard_id (:id dashboard))
after
(drop-excessive-nested! ingested dashboard :tags)
```

Reply:
In theory this makes it possible to write `sync-nested!` function, which will take `[key ingested local make-fn]`. OTOH it could be obscuring the process a bit which will make writing `make-fn` a bit harder (it needs you to understand what's going on).

Reply:
`sync-nested!` is somewhat harder, since we could have empty `local-nested` and then how do I determine model for insertion?.. I think I'll leave separate loop intact for now.